test/dlc_tests/sanity/test_pre_release.py

```diff
@@ -150,26 +150,31 @@ def test_framework_and_cuda_version_gpu(gpu, ec2_connection):
     :param ec2_connection: fixture to establish connection with an ec2 instance
     """
     image = gpu
-    if "tensorflow-inference" in image:
-        pytest.skip(msg="TF inference does not have core tensorflow installed")
-
     tested_framework, tag_framework_version = get_framework_and_version_from_tag(image)
 
-    # Module name is "torch"
-    if tested_framework == "pytorch":
-        tested_framework = "torch"
-    cmd = f"import {tested_framework}; print({tested_framework}.__version__)"
-    output = ec2.execute_ec2_training_test(ec2_connection, image, cmd, executable="python")
-
-    if is_canary_context():
-        assert tag_framework_version in output.stdout.strip()
-    else:
-        assert tag_framework_version == output.stdout.strip()
+    # Framework Version Check #
+    # Skip framework version test for tensorflow-inference, since it doesn't have core TF installed
+    if "tensorflow-inference" not in image:
+        # Module name is "torch"
+        if tested_framework == "pytorch":
+            tested_framework = "torch"
+        cmd = f"import {tested_framework}; print({tested_framework}.__version__)"
+        output = ec2.execute_ec2_training_test(ec2_connection, image, cmd, executable="python")
+
+        if is_canary_context():
+            assert tag_framework_version in output.stdout.strip()
+        else:
+            assert tag_framework_version == output.stdout.strip()
```

Comment:
Question: Why is there a difference in the output of canary vs others?

Reply:
Canary images generally take the major.minor version instead of major.minor.patch due to the fact that they use SM tags. This was not part of this revision, i just moved all of the framework version test part into an if condition so it looks like i added this